mojo/stdlib/stdlib/hashlib/_ahash.mojo

```diff
@@ -125,47 +126,38 @@ struct AHasher[key: U256](Defaultable, Hasher):
         var combined = _folded_multiply(xored[0], xored[1])
         self.buffer = rotate_bits_left[ROT]((self.buffer + self.pad) ^ combined)
 
-    fn _update_with_bytes(
-        mut self,
-        data: UnsafePointer[
-            UInt8, address_space = AddressSpace.GENERIC, mut=False, **_
-        ],
-        length: Int,
-    ):
-        """Consume provided data to update the internal buffer.
-
-        Args:
-            data: Pointer to the byte array.
-            length: The length of the byte array.
+    fn _update_with_bytes(mut self, data: Span[UInt8]):
         """
+        Consume provided data to update the internal buffer.
+        """
+        var length = len(data)
+        var ptr = data.unsafe_ptr()
         self.buffer = (self.buffer + length) * MULTIPLE
         if length > 8:
             if length > 16:
                 var tail = (
-                    data.offset(length - 16)
+                    (ptr + (length - 16))
```

Comment:
```suggestion
                    (ptr[length-16])
```
i.e. use the `UnsafePointer.__getitem__` instead of naked pointer arithmetic.  Ditto below.